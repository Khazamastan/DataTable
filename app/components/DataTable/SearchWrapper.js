import styled from 'styled-components';
const SearchWrapper = styled.div`
  margin-bottom: 20px;
  display: flex;
  overflow: hidden;
  > input {
    flex: 1 0 auto;
    border-radius: 0 3px 3px 0;
  }
  > div {
    flex-basis: 100px;
    > button {
      width: 100%;
      border: 1px solid ${props => props.theme.borderColor};
      border-radius: 3px 0 0 3px;
    }
  }
`;

export default SearchWrapper;
