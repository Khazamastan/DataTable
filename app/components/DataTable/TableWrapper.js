import styled from 'styled-components';

const TableWrapper = styled.div`
  font-family: arial, sans-serif;
  width: 100%;
`;

export default TableWrapper;
