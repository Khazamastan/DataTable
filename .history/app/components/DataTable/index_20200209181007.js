import styled from 'styled-components';
import React, { useState, useMemo, useEffect } from 'react';
import TableWrapper from './TableWrapper';
import TableHead from '../TableHead';
import TableBody from '../TableBody';
import _ from 'lodash';

const Table = props => {
  const { columns, data } = props;
  let noResultsConent;
  const stateData = {
    selectedRowIds: {},
    allSelected: true,
    selectedRowsOriginal: {},
  };

  const [selectedRowIds, setSelectedRowIds] = useState(
    stateData.selectedRowIds,
  );

  const [allSelected, setAllSelected] = useState(allSelected);

  const getRowIsSelected = id => selectedRowIds[id];

  const selectRow = (row, all, status) => {
    const selectedMap = {};
    if (all) {
      data.forEach(row => {
        selectedMap[row.id] = status;
      });
      setAllSelected(status);
    } else {
      const { id } = row;
      const isSelected = getRowIsSelected(id);
      selectedMap[id] = !isSelected;
      setAllSelected(false);
    }

    Object.assign(selectedRowIds, selectedMap);
    setSelectedRowIds(selectedRowIds);
    console.log(selectedRowIds);
    myref && myref.forceUpdateGrid();
  };

  const selectedRowsOriginal = useEffect(() => {
    const selectedRowsOriginal = [];
    data.forEach(row => {
      const isSelected = getRowIsSelected(row.id);
      if (isSelected) {
        selectedRowsOriginal.push(row);
      }
    });
    return selectedRowsOriginal;
  }, [selectedRowIds]);
  let myref;
  const setMyRef = (ref) => {
    myref= ref;
  }
  if (!data.length) {
    noResultsConent = (
      <p className="no-results" key="only">
        No Results
      </p>
    );
  }

  return (
    <div className="container">
      {selectedRowsOriginal && selectedRowsOriginal.length}
      <div className="row">
        <TableWrapper>
          <TableHead
            columns={columns}
            onSelectAll={selectRow}
            allSelected={allSelected}
            selectedRowsOriginal={selectedRowsOriginal}
          />
          <TableBody
            columns={columns}
            setMyRef={setMyRef}
            onRowClick={props.onRowClick}
            selectedRowIds={selectedRowIds}
            onSelectRow={selectRow}
            allSelected={allSelected}
            data={data}
          />
        </TableWrapper>
        {noResultsConent}
      </div>
    </div>
  );
};

export default Table;
