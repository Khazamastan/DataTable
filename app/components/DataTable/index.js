import styled from 'styled-components';
import React from 'react';
import TableWrapper from './TableWrapper';
import TableHead from '../TableHead';
import TableBody from '../TableBody';

const Table = props => {
  const { columns, data } = props;
  let noResultsConent;
  if (!data.length) {
    noResultsConent = (
      <p className="no-results" key="only">
        No Results
      </p>
    );
  }
  return (
    <div className="container">
      <div className="row">
        <TableWrapper>
          <TableHead columns={columns} />
          <TableBody
            columns={columns}
            onRowClick={props.onRowClick}
            onSelectRow={props.onSelectRow}
            data={data}
          />
        </TableWrapper>
        {noResultsConent}
      </div>
    </div>
  );
};

export default Table;
