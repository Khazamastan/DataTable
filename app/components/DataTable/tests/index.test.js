/**
 * Test the DataTable
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import LanguageProvider from 'containers/LanguageProvider';
import DataTable from 'components/DataTable';
import { browserHistory } from 'react-router-dom';
// eslint-disable-next-line no-unused-vars
import { debounce } from 'lodash';
import { translationMessages } from '../../../i18n';
import configureStore from '../../../configureStore';
const sampleState = {
  data: [],
  columns: [
    {
      key: 'testCell1',
      label: 'name',
    },
    {
      key: 'testCell2',
      label: 'age',
    },
  ],
  onRowClickHandler: () => {},
  onSelectRowHandler: () => {},
};
const renderComponent = (props = {}) =>
  render(
    <DataTable
      data={props.data}
      query={{}}
      loading={null}
      loadMoreRows={() => {}}
      columns={props.columns}
      onRowClick={props.onRowClickHandler}
      onSelectRow={props.onSelectRowHandler}
    />,
  );

describe('<DataTable />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <LanguageProvider messages={translationMessages}>
          <IntlProvider locale="en" defaultLocale="en">
            <DataTable
              columns={[]}
              data={[]}
              query={{}}
              loading={null}
              loadMoreRows={() => {}}
              totalCount={0}
              onRowClick={() => {}}
              onSelectRow={() => {}}
            />
          </IntlProvider>
        </LanguageProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  describe('test No data case', () => {
    describe('shoud show No Data Text', () => {
      sampleState.data = null;
      const { container } = renderComponent(sampleState);
      expect(container.querySelector('.no-results')).toBeDefined();
      expect(container.querySelector('.no-results').textContent).toEqual(
        'No Data',
      );
    });

    describe('shoud not show No Data Text', () => {
      sampleState.data = [{}];
      const { container } = renderComponent(sampleState);
      expect(container.querySelector('.no-results')).toBeNull();
    });
  });

  describe('test with data case', () => {
    sampleState.data = [
      {
        name: 'testValue',
        age: 20,
      },
    ];
    describe('shoud show Showing text', () => {
      const { container } = renderComponent(sampleState);
      expect(container.querySelector('.showing-text')).toBeDefined();
      expect(container.querySelector('.showing-text').textContent).toEqual(
        'Showing 1',
      );
    });

    describe('shoud show two columns headers', () => {
      const { container } = renderComponent(sampleState);
      expect(container.querySelector('.testCell1')).toBeDefined();
      expect(container.querySelector('.testCell1').textContent).toEqual(
        'name ',
      );

      expect(container.querySelector('.testCell2')).toBeDefined();
      expect(container.querySelector('.testCell2').textContent).toEqual('age ');
    });
  });
});
