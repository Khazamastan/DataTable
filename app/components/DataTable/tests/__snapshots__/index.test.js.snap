// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataTable /> should render and match the snapshot 1`] = `
<div
  class="container"
>
  <div
    class="SearchWrapper-q18sbo-0 fjVACF"
  >
    <div
      class="Wrapper-sc-1yw1li4-0 efyKhh"
    >
      <button
        class="StyledButton-sc-1lrmbv8-0 lkmlui"
      >
        Filter
      </button>
    </div>
    <input
      class="search Wrapper__InputWrapper-sc-15n0e2a-0 joPiQc"
      placeholder="Search here"
      type="text"
      value=""
    />
  </div>
  <p
    class="showing-text"
  >
    Showing 
    <strong>
      0
    </strong>
  </p>
  <div
    class="TableWrapper-sc-1eiqmgm-0 kWvpHK"
  >
    <div
      class="Wrapper-sc-12b7pcm-0 evwzjo"
    >
      <div />
    </div>
    <div
      class="Wrapper-sc-1766fc-0 joDhtK"
    >
      <div>
        <div
          style="overflow: visible; width: 0px;"
        >
          <div>
            <div
              aria-label="grid"
              aria-readonly="true"
              class="ReactVirtualized__Grid ReactVirtualized__List"
              role="grid"
              style="box-sizing: border-box; direction: ltr; height: auto; position: relative; width: 0px; will-change: transform; overflow-x: hidden; overflow-y: hidden;"
              tabindex="0"
            />
          </div>
        </div>
        <div
          class="resize-triggers"
        >
          <div
            class="expand-trigger"
          >
            <div
              style="width: 1px; height: 1px;"
            />
          </div>
          <div
            class="contract-trigger"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
